diff -Nru mybatis-3-mybatis-3.2.8/pom.xml mybatis-3-mybatis-3.2.8-gil/pom.xml
--- mybatis-3-mybatis-3.2.8/pom.xml	2014-10-10 18:44:34.000000000 +0200
+++ mybatis-3-mybatis-3.2.8-gil/pom.xml	2014-12-26 14:27:03.038441294 +0100
@@ -136,9 +136,9 @@
 
   <dependencies>
     <dependency>
-      <groupId>ognl</groupId>
-      <artifactId>ognl</artifactId>
-      <version>2.6.9</version>
+      <groupId>org.apache.commons</groupId>
+      <artifactId>commons-ognl</artifactId>
+      <version>4.0-incubating-SNAPSHOT</version>
       <scope>provided</scope>
       <optional>true</optional>
     </dependency>
diff -Nru mybatis-3-mybatis-3.2.8/src/main/java/org/apache/ibatis/scripting/xmltags/DynamicContext.java mybatis-3-mybatis-3.2.8-gil/src/main/java/org/apache/ibatis/scripting/xmltags/DynamicContext.java
--- mybatis-3-mybatis-3.2.8/src/main/java/org/apache/ibatis/scripting/xmltags/DynamicContext.java	2014-10-10 18:44:34.000000000 +0200
+++ mybatis-3-mybatis-3.2.8-gil/src/main/java/org/apache/ibatis/scripting/xmltags/DynamicContext.java	2014-12-26 14:30:00.065088895 +0100
@@ -18,9 +18,9 @@
 import java.util.HashMap;
 import java.util.Map;
 
-import ognl.OgnlException;
-import ognl.OgnlRuntime;
-import ognl.PropertyAccessor;
+import org.apache.commons.ognl.OgnlException;
+import org.apache.commons.ognl.OgnlRuntime;
+import org.apache.commons.ognl.MapPropertyAccessor;
 
 import org.apache.ibatis.reflection.MetaObject;
 import org.apache.ibatis.session.Configuration;
@@ -107,7 +107,7 @@
     }
   }
 
-  static class ContextAccessor implements PropertyAccessor {
+  static class ContextAccessor extends MapPropertyAccessor {
 
     public Object getProperty(Map context, Object target, Object name)
         throws OgnlException {
diff -Nru mybatis-3-mybatis-3.2.8/src/main/java/org/apache/ibatis/scripting/xmltags/OgnlCache.java mybatis-3-mybatis-3.2.8-gil/src/main/java/org/apache/ibatis/scripting/xmltags/OgnlCache.java
--- mybatis-3-mybatis-3.2.8/src/main/java/org/apache/ibatis/scripting/xmltags/OgnlCache.java	2014-10-10 18:44:34.000000000 +0200
+++ mybatis-3-mybatis-3.2.8-gil/src/main/java/org/apache/ibatis/scripting/xmltags/OgnlCache.java	2014-12-26 14:31:45.179409686 +0100
@@ -19,8 +19,8 @@
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
 
-import ognl.Ognl;
-import ognl.OgnlException;
+import org.apache.commons.ognl.Ognl;
+import org.apache.commons.ognl.OgnlException;
 
 import org.apache.ibatis.builder.BuilderException;
 
diff -Nru mybatis-3-mybatis-3.2.8/src/main/java/org/apache/ibatis/scripting/xmltags/OgnlClassResolver.java mybatis-3-mybatis-3.2.8-gil/src/main/java/org/apache/ibatis/scripting/xmltags/OgnlClassResolver.java
--- mybatis-3-mybatis-3.2.8/src/main/java/org/apache/ibatis/scripting/xmltags/OgnlClassResolver.java	2014-10-10 18:44:34.000000000 +0200
+++ mybatis-3-mybatis-3.2.8-gil/src/main/java/org/apache/ibatis/scripting/xmltags/OgnlClassResolver.java	2014-12-26 14:32:53.568668180 +0100
@@ -19,7 +19,7 @@
 import java.util.HashMap;
 import java.util.Map;
 
-import ognl.ClassResolver;
+import org.apache.commons.ognl.ClassResolver;
 
 import org.apache.ibatis.io.Resources;
 
